src/test/kotlin/app/batch/MessageProducerTest.kt

```diff
@@ -0,0 +1,273 @@
+package app.batch
+
+import app.domain.DataKeyResult
+import app.domain.EncryptionResult
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.Appender
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import junit.framework.Assert.assertEquals
+import org.everit.json.schema.loader.SchemaLoader
+import org.json.JSONObject
+import org.json.JSONTokener
+import org.junit.Test
+import org.slf4j.LoggerFactory
+
+class MessageProducerTest {
+
+    @Test
+    fun testValidObjectGivesSchemaValidMessage() {
+        val dateKey = "\$date"
+        val dateValue = "2018-12-14T15:01:02.000+0000"
+        val idFieldValue = "idFieldValue"
+        val anotherIdFieldValue = "anotherIdFieldValue"
+
+        val validJson = """{
+            "_id": {
+                "idField": "$idFieldValue",
+                "anotherIdField": "$anotherIdFieldValue"
+            },
+            "_lastModifiedDateTime": {
+                "$dateKey": "$dateValue" 
+            }
+        }""".trimIndent()
+
+        val parser: Parser = Parser.default()
+        val stringBuilder = StringBuilder(validJson)
+        val jsonObject = parser.parse(stringBuilder) as JsonObject
+        val initialisationVector = "initialisationVector"
+        val encrypted = "encrypted"
+        val encryptionResult = EncryptionResult(initialisationVector, encrypted)
+
+        val dataKeyEncryptionKeyId = "cloudhsm:1,2"
+        val plaintextDataKey = "plaintextDataKey"
+        val ciphertextDataKey = "ciphertextDataKey"
+        val dataKeyResult = DataKeyResult(dataKeyEncryptionKeyId, plaintextDataKey, ciphertextDataKey)
+        val database = "database"
```

Comment:
If these are fixed in lots of tests lets have them as member vars, to reduce the boilerplate in each test?

Reply:
Done.